commit 2b3e068a598febc7e811344d9a6a48589f4f5a4e
Author: Jeff Law <jlaw@ventanamicro.com>
Date:   Wed Oct 9 09:18:12 2024 -0600

    Revert "RISC-V: Add implication for M extension."
    
    This reverts commit 0a193466f2e87acef9b86e0d086bc6f6017518b0.

diff --git a/gcc/common/config/riscv/riscv-common.cc b/gcc/common/config/riscv/riscv-common.cc
index 60595a3e356..2adebe0b6f2 100644
--- a/gcc/common/config/riscv/riscv-common.cc
+++ b/gcc/common/config/riscv/riscv-common.cc
@@ -75,8 +75,6 @@ struct riscv_implied_info_t
 /* Implied ISA info, must end with NULL sentinel.  */
 static const riscv_implied_info_t riscv_implied_info[] =
 {
-  {"m", "zmmul"},
-
   {"d", "f"},
   {"f", "zicsr"},
   {"d", "zicsr"},
commit b977520403052ed43a72378bf64d8af0a14d72ff
Author: Jeff Law <jlaw@ventanamicro.com>
Date:   Wed Oct 9 10:45:47 2024 -0600

    Revert "RISC-V: Enable builtin __riscv_mul with Zmmul extension."
    
    This reverts commit 2990f5802a727cbd717587c3a345fa940193049f.

diff --git a/gcc/config/riscv/riscv-c.cc b/gcc/config/riscv/riscv-c.cc
index 7e9c478e97b..71112d9c66d 100644
--- a/gcc/config/riscv/riscv-c.cc
+++ b/gcc/config/riscv/riscv-c.cc
@@ -123,7 +123,7 @@ riscv_cpu_cpp_builtins (cpp_reader *pfile)
   if (TARGET_ATOMIC)
     builtin_define ("__riscv_atomic");
 
-  if (TARGET_ZMMUL)
+  if (TARGET_MUL)
     builtin_define ("__riscv_mul");
   if (TARGET_DIV)
     builtin_define ("__riscv_div");
